### ISSUE TITLE
Multiple triggers of the gitStream workflow for the same PR

### ISSUE THREAD

**Issue by yakirm-cr** [linear-b/gitstream#101]
**Describe the bug**
Every time a PR is synchronised (via rebase or new commits), the gitStream workflow is triggered via the GitHub app (bot) more than once.
As we we have multiple PRs open in parallel, the gitStream workflow executions fill the "Actions" tab and also consume more GitHub actions minutes.

**To Reproduce**

1. Open a new PR.
2. Synchronise it (via rebase or new commits).

**Expected behavior**
The gitStream workflow should be triggered only once per synchronise event.

**Screenshots**
I selected the gitStream workflow specifically in the "Actions" tab in order to show duplicate triggers in the same time:
![image](https://user-images.githubusercontent.com/86773995/232498799-50375d94-ba0e-4ee3-9069-787369632bb2.png)

**Additional context**
N/A

**Comment by vim-zz** [comment]
@yakirm-cr gitStream should trigger once for every commit. In addition if you have CM rules that use the `pr` context it will also trigger on PR title and description edits. 

1. do you use the `pr` context in your CM file? 
2. do you see more triggers than described above? if **yes**, can you open a new PR (with simple change, e.g. README file), and see if you get only a single GitHub action trigger (as it should) or you get multiple (a bug)

**Comment by yakirm-cr** [comment]
@vim-zz 
Thanks for the clarification. :slightly_smiling_face: 

> 
> 1. do you use the `pr` context in your CM file?

Yes, we use the `pr` context in one of our CM files.
We use GitHub dependabot for auto-updating dependencies and dependabot constantly updates the PR description (when rebasing PRs), which explains the multiple triggers of gitStream workflow for each PR opened by dependabot. :slightly_smiling_face: 

> 2. do you see more triggers than described above? if **yes**, can you open a new PR (with simple change, e.g. README file), and see if you get only a single GitHub action trigger (as it should) or you get multiple (a bug)

For PRs opened by developers we do not encounter this issue.
So, thanks again for the clarification!
